nile-backend/src/main/java/com/tp/Nile/services/ProductServiceImpl.java

```diff
@@ -65,7 +68,58 @@ public Product getProductById(Integer productId) throws InvalidProductIdExceptio
 
     }
 
-    public Product addProduct(Product product) {
+    public Product addProduct(Product product) throws MissingPropertyException {
```

Comment:
Are you sure this logic is necessary? Did you write any tests before writing this?

Reply:
When just changing the cascade types and not adding additional logic, the same error occurs